### DISCUSSION TITLE
How would I go about making a conveyor belt for the character to walk on

### DISCUSSION THREAD

**Discussion by ckdlskgkdrhd** [naverz/zepeto-studio-global#36] [Q&A]
I've been pondering on how I'd go about making a conveyor belt for my map.
I've tried transform.Translate(x,y,z) but that'd simply teleport the character to the designated xyz location.
I've tried the built-in MoveToPosition, MoveContinuously which works similarly to what I've trying to create minus the running animation.
now I'm outta ideas.

**Comment by naverz-mkao** [comment]
Hi!

CharacterController class has a "Move()" function you can use to achieve moving the character without playing its animations.

Also, if you are curious, take a look at the world-movingblock-sample project. This is how you implement moving blocks that will also move the player character with it (like in Jump Master). It also uses CharacterController.Move().

[https://github.com/naverz/world-movingblock-sample/blob/main/Assets/Scripts/MovingBlock.ts](url)

Below is a code sample of how to use CharacterController.Move() to move your player character without playing animations.

```import { CharacterController, Input, KeyCode, Time, Vector3 } from 'UnityEngine'
import { ZepetoCharacter, ZepetoPlayers } from 'ZEPETO.Character.Controller';
import { ZepetoScriptBehaviour } from 'ZEPETO.Script'

export default class MoveCharacterTest extends ZepetoScriptBehaviour {
    private localCharacter: ZepetoCharacter;
    private localCharacterController: CharacterController;

    Start() {    
        ZepetoPlayers.instance.OnAddedLocalPlayer.AddListener(() => {
            this.localCharacter = ZepetoPlayers.instance.LocalPlayer.zepetoPlayer.character;
            this.localCharacterController = this.localCharacter.GetComponent<CharacterController>();
        });
    }

    Update() {
        if (Input.GetKey(KeyCode.T)) {

            this.localCharacterController.Move(new Vector3(1 * Time.deltaTime,0,0));
        }
    }

}

**Reply by ckdlskgkdrhd** [reply]
@naverz-mkao thanks a lot!. I'll definitely take a look into it.

edit: I got it working perfectly. :D
